src/core/Akka/Actor/Internal/ActorSystemImpl.cs

```diff
@@ -177,10 +177,8 @@ private void ConfigureExtensions(IEnumerable<IExtensionId> extensionIdProviders)
         public override object RegisterExtension(IExtensionId extension)
         {
             if(extension == null) return null;
-            if(!_extensions.ContainsKey(extension.ExtensionType))
-            {
-                _extensions.TryAdd(extension.ExtensionType, new Lazy<object>(() => extension.CreateExtension(this)));
-            }
```

Comment:
This part was the cause for race conditions in the old code.
there is a check if something does not exist, and if so, it adds an extension.. classic race condition semantics.

The new code fixes that.